src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -711,6 +711,10 @@ bool IsInterfaceOverrideThatDoesNotNeedMarked (OverrideInformation overrideInfor
 			if (!overrideInformation.IsOverrideOfInterfaceMember || isInstantiated)
 				return false;
 
+			// This is a static interface method
+			if (overrideInformation.Override.IsStatic && overrideInformation.Base.IsStatic)
```

Comment:
I would personally make this check as specific as possible - I'd check that the base is abstract and the override is non-virtual (I believe that's always the case unless I misread the spec).